doc/apiary/v2/fiware-ngsiv2-reference.apib

```diff
@@ -913,6 +923,21 @@ Note that if a custom payload is used for the notification (the field `payload`
 corresponding subscription), then a value of `custom` is used for the `Ngsiv2-AttrsFormat` header
 in the notification.
 
+## Notification metadata marks
+
+In the case of using `metadataFlags` to `true` in the subscription, the following metadata
+are automatically inserted in the notification messages:
+
+* For these attributes included in the update triggering the notification, no matter if the actually
+  changed or not, `ngsi:onArrival` with boolean value `true` and type `"Boolean"` is inserted.
+
+* For these attributes included in the update triggering the notification that actually changed their
+  values, `ngsi:onChange` with boolean value `true` and type `"Boolean"` is inserted.
+
+* For these attributes notififed as a consecuence of an initial creation or update of the
+  subscription, `ngsi:onSubscriptionChange` with boolean value `true` and type `"Boolean"`
+  is inserted.
+
```

Comment:
Just to see that I have understood this correctly.

Notifications due to attribute value update:
  Updated (triggering) attributes will have a metadata called ngsi:onChange set to TRUE
  All other included attributes will have a metadata called ngsi:onArrival

Notifications due to sub creation or update:
  ALL attributes will have a metadata called ngsi:onSubscriptionChange

If this is the case, it seems extremely verbose.
What if we have 1000 attributes?
Add a metadata (50 chars?) for each and every attribute?
 
Why not simply add a field to the Notification itself, informing about dueToAttrValueUpdate/dueToSubscriptionChange - valid for all attributes.
And then, the triggering attributes should have a metadata indicating that they are 'triggering'. The non-triggering attributes need no metadata at all. The absence of the metadata already indicates they're not triggering attributes.